app/scripts/metamask-controller.js

```diff
@@ -330,12 +329,24 @@ export default class MetamaskController extends EventEmitter {
         this.platform.showTransactionNotification(txMeta, rpcPrefs);
 
         const { txReceipt } = txMeta;
+        const metamaskState = await this.getState();
+
         if (txReceipt && txReceipt.status === '0x0') {
-          this.sendBackgroundMetaMetrics({
-            action: 'Transactions',
-            name: 'On Chain Failure',
-            customVariables: { errorMessage: txMeta.simulationFails?.reason },
-          });
+          this.metaMetricsController.trackEvent(
+            {
+              category: 'Background',
+              properties: {
+                action: 'Transactions',
+                errorMessage: txMeta.simulationFails?.reason,
+                numberOfTokens: metamaskState.tokens?.length ?? 0,
+                numberOfAccounts: Object.keys(metamaskState.accounts ?? {})
```

Comment:
Same as above with `accounts` and `{}`

Reply:
Same as above, but here I did add something that didn't exist before. I'll check into both of these and simplify the selectors if they can be.